core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -464,17 +466,33 @@ private void verifyResultsWithPushdownDisabled()
 
         private void assertPlan(PlanMatchPattern expectedPlan)
         {
-            transaction(runner.getTransactionManager(), runner.getAccessControl())
-                    .execute(session, session -> {
-                        Plan plan = runner.createPlan(session, query, WarningCollector.NOOP);
-                        PlanAssert.assertPlan(
-                                session,
-                                runner.getMetadata(),
-                                runner.getFunctionManager(),
-                                noopStatsCalculator(),
-                                plan,
-                                expectedPlan);
-                    });
+            try {
+                transaction(runner.getTransactionManager(), runner.getAccessControl())
+                        .execute(session, session -> {
+                            Plan plan = runner.createPlan(session, query, WarningCollector.NOOP);
+                            PlanAssert.assertPlan(
+                                    session,
+                                    runner.getMetadata(),
+                                    runner.getFunctionManager(),
+                                    noopStatsCalculator(),
+                                    plan,
+                                    expectedPlan);
+                        });
+            }
+            catch (AssertionError e) {
+                // Catch the plan assertion error to add our description to it
+                throw descriptionText().isEmpty()
+                        ? e
+                        : new AssertionError(format("%s: %s", descriptionText(), e.getMessage()), e.getCause());
+            }
+        }
+
+        /**
+         * Get this assertion's description if it exists, or the provided description otherwise.
+         */
+        private String defaultDescription(String fallbackFormat, Object... args)
+        {
+            return descriptionText().isEmpty() ? format(fallbackFormat, args) : descriptionText();
```

Comment:
Instead of supporting this (which is cumbersome and error-prone)
i'd rather disallow `as()` method.